### PR TITLE
Fix TierCalculationTest that might fail on slow builds

### PR DESCRIPTION
**Pull request by henri-tremblay**

### CODE REVIEW

**Comment by ljacomet** [comment]
test this please

For some reason, no build was triggered

**Comment by ljacomet** [comment]
test this please

**Comment by henri-tremblay** [comment]
Interesting. It failed but on a different thing. It really feels like the entry was evicted. I will dig a bit more.

**Comment by henri-tremblay** [comment]
test this please

**Comment by henri-tremblay** [comment]
test this please

File: integration-test/src/test/java/org/ehcache/integration/statistics/TierCalculationTest.java [hunk 1]
```diff
@@ -328,7 +328,7 @@ public void testEviction() {
   @Test
   public void testExpiration() throws InterruptedException {
     cache.put(1, "a");
-    Thread.sleep(100);
+    Thread.sleep(300);
```
Comment (ljacomet) [inline integration-test/src/test/java/org/ehcache/integration/statistics/TierCalculationTest.java]: huh? please inject a `TimeSource` that enables you to simulate time - that `sleep` needs to go

Comment (henri-tremblay) [inline integration-test/src/test/java/org/ehcache/integration/statistics/TierCalculationTest.java]: Wasn't aware of the `TimeSource`. Much better solution. Will do.

File: impl/src/main/java/org/ehcache/impl/internal/DefaultTimeSourceService.java [hunk 2]
```diff
@@ -29,6 +29,14 @@
 
   private final TimeSource timeSource;
 
+  public DefaultTimeSourceService() {
+    this.timeSource = SystemTimeSource.INSTANCE;
+  }
+
+  public DefaultTimeSourceService(TimeSource timeSource) {
```
Comment (ljacomet) [inline impl/src/main/java/org/ehcache/impl/internal/DefaultTimeSourceService.java]: Not needed, see following comment

File: integration-test/src/test/java/org/ehcache/integration/statistics/TierCalculationTest.java [hunk 3]
```diff
@@ -68,6 +74,7 @@ public void before() throws Exception {
       .withCache("cache", cacheConfiguration)
       .using(new DefaultPersistenceConfiguration(diskPath.newFolder()))
       .using(statisticsService)
+      .using(new DefaultTimeSourceService(timeSource))
```
Comment (ljacomet) [inline integration-test/src/test/java/org/ehcache/integration/statistics/TierCalculationTest.java]: Just use an anonymous inner `TimeSourceService` to hand off the  `TestTimeSource` and then you do not need to add the injection constructor on the default one.

Comment (henri-tremblay) [inline integration-test/src/test/java/org/ehcache/integration/statistics/TierCalculationTest.java]: Will pass the `TimeSourceServiceConfig` instead

**Comment by ljacomet** [comment]
Unstable test

**Pull request merged**